core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java

```diff
@@ -105,15 +110,27 @@ public void statusChanged(boolean login) {
           HttpRequestInitializer initializer = new HttpRequestInitializer() {
             @Override
             public void initialize(HttpRequest httpRequest) throws IOException {
+              HttpHeaders headers = new HttpHeaders();
               httpRequest.setConnectTimeout(timeout);
               httpRequest.setReadTimeout(timeout);
+              httpRequest.setHeaders(headers);
               credential.initialize(httpRequest);
             }
           };
 
           HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-          cloudDebuggerClient = new Builder(httpTransport, JSON_FACTORY, initializer).setRootUrl(ROOT_URL)
-            .setApplicationName(GoogleLoginUtils.getCurrentPlatformName()).build().debugger();
+          String userAgent = USER_AGENT;
+          try {
+            userAgent += GoogleLoginUtils.getCurrentPlatformName()
```

Comment:
Maybe common-lib?

I guess it doesn't have to be static. I wouldn't object to that.

On Thu, Jan 7, 2016 at 5:31 PM, Elliotte Rusty Harold <
notifications@github.com> wrote:

> In core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/327#discussion_r49135850>
> :
>
> >                credential.initialize(httpRequest);
> >              }
> >            };
> >
> >            HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
> > -          cloudDebuggerClient = new Builder(httpTransport, JSON_FACTORY, initializer).setRootUrl(ROOT_URL)
> > -            .setApplicationName(GoogleLoginUtils.getCurrentPlatformName()).build().debugger();
> > +          String userAgent = USER_AGENT;
> > +          try {
> > +            userAgent += GoogleLoginUtils.getCurrentPlatformName()
>
> Sounds good, though I do wish we could inject it rather than making it
> static. Where would you suggest? Somewhere in the login package?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/327/files#r49135850>
> .
>